src/stactools/core/addraster.py

```diff
@@ -25,27 +27,29 @@ def add_raster_to_item(item: Item) -> Item:
     for asset in item.assets.values():
         if asset.roles and "data" in asset.roles:
             raster = RasterExtension.ext(asset)
-            bands = []
             href = make_absolute_href(asset.href, item.get_self_href())
-            dataset = gdal.Open(href, gdal.GA_ReadOnly)
-            for nband in range(dataset.RasterCount):
-                gdal_band = dataset.GetRasterBand(nband + 1)
-                band = RasterBand.create()
-                band.nodata = gdal_band.GetNoDataValue()
-                band.spatial_resolution = dataset.GetGeoTransform()[1]
-                band.data_type = DataType(
-                    gdal.GetDataTypeName(gdal_band.DataType).lower())
-                minimum = gdal_band.GetMinimum()
-                maximum = gdal_band.GetMaximum()
-                if not minimum or not max:
-                    minimum, maximum = gdal_band.ComputeRasterMinMax(True)
-                band.statistics = Statistics.create(minimum=minimum,
-                                                    maximum=maximum)
-                hist_data = gdal_band.GetHistogram(minimum, maximum,
-                                                   NUM_BUCKETS)
-                band.histogram = Histogram.create(NUM_BUCKETS, minimum,
-                                                  maximum, hist_data)
-                bands.append(band)
+            bands = _read_bands(href)
             if bands:
                 raster.apply(bands)
     return item
+
+
+def _read_bands(href: str) -> List[RasterBand]:
+    bands = []
+    with rasterio.open(href) as dataset:
+        for (i, index) in enumerate(dataset.indexes):
+            data = dataset.read(index, masked=True)
+            band = RasterBand.create()
+            band.nodata = dataset.nodatavals[i]
+            band.spatial_resolution = dataset.transform[0]
+            band.data_type = DataType(dataset.dtypes[i])
+            minimum = float(numpy.min(data))  # type: ignore
+            maximum = float(numpy.max(data))  # type: ignore
```

Comment:
```suggestion
            minimum = float(numpy.min(data))
            maximum = float(numpy.max(data))
```

I get `error: Unused "type: ignore" comment` when I run the lint script locally for these lines.

It looks like `scripts/lint` doesn't get run as part of the CI, is that intentional?

Reply:
We hit `scripts/cibuild` [here](https://github.com/stac-utils/stactools/blob/08f61bdf5f51db0dec4dbbeb7472a7b15091d722/.github/workflows/continuous-integration.yml#L58), which ends up calling `scripts/lint` [here](https://github.com/stac-utils/stactools/blob/08f61bdf5f51db0dec4dbbeb7472a7b15091d722/scripts/test#L20). I needed those `type: ignore` slugs to get things passing CI with our environment. Do you have more **numpy** typing available in your environment? If so can you point me at it? :-D

Reply:
> We hit `scripts/cibuild` [here](https://github.com/stac-utils/stactools/blob/08f61bdf5f51db0dec4dbbeb7472a7b15091d722/.github/workflows/continuous-integration.yml#L58), which ends up calling `scripts/lint` [here](https://github.com/stac-utils/stactools/blob/08f61bdf5f51db0dec4dbbeb7472a7b15091d722/scripts/test#L20). I needed those `type: ignore` slugs to get things passing CI with our environment. Do you have more **numpy** typing available in your environment? If so can you point me at it? :-D

Oops, sorry. I missed the call in `scripts/test` 🤦🏻 .

Looks like it's a difference in the `numpy` version. I was running `numpy==1.22.1` and didn't need those ignore statements. Going back to `numpy==1.21.1` (which is what is installed in the `conda` environment in the CI container) gives me errors without those ignore comments.